### ISSUE TITLE
lzma library not detected on mac

### ISSUE THREAD

**Issue by sansuiso** [pwalch/lonesnake#2]
## Python import crash

When installing python 3.10 via `lonesnake --py 3.10` then installing PyTorch's `torchvision` module in the environment I get the following error at import:

```
Python 3.10.11 (main, Apr  8 2023, 15:30:41) [Clang 14.0.3 (clang-1403.0.22.14.1)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import torchvision
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<REDACTED>/.lonesnake/venv/lib/python3.10/site-packages/torchvision/__init__.py", line 6, in <module>
    from torchvision import datasets, io, models, ops, transforms, utils
  File "<REDACTED>/.lonesnake/venv/lib/python3.10/site-packages/torchvision/datasets/__init__.py", line 1, in <module>
    from ._optical_flow import FlyingChairs, FlyingThings3D, HD1K, KittiFlow, Sintel
  File "<REDACTED>/.lonesnake/venv/lib/python3.10/site-packages/torchvision/datasets/_optical_flow.py", line 13, in <module>
    from .utils import _read_pfm, verify_str_arg
  File "<REDACTED>/.lonesnake/venv/lib/python3.10/site-packages/torchvision/datasets/utils.py", line 6, in <module>
    import lzma
  File "<REDACTED>/.lonesnake/interpreter/usr/local/lib/python3.10/lzma.py", line 27, in <module>
    from _lzma import *
ModuleNotFoundError: No module named '_lzma'
```

Long story short, some utilities from `torchvision`rely on `lzma.py` which itself is only properly instantiated if the corresponding library can be found when building python.

## Possible fixes

On macOS `liblzma.dylib` is installed via `brew install xz`, but this is not enough one must pass the corresponding path to the `configure` invocation.

### Python 3.10

Sadly there is no option to pass a path to LZMA with python 3.10. The trick here is to add it to the command via the env variable `CPP_FLAGS` like in `CPPFLAGS="-I$(brew --prefix xz)/include" lonesnake --py 3.10`.

### Python 3.11

Starting with 3.11 the configure file allows to set variables for LZMA specifcally and the following command should work (to be confirmed):
```
LIBLZMA_CFLAGS="$(brew --prefix xz)/include" LIBLZMA_LIBS="-L $(brew --prefix xz)/lib -llzma" ./configure "$with_openssl_str"
```

**Comment by pwalch** [comment]
@sansuiso Thanks for reporting the issue! 

I was able to reproduce this on macOS Ventura 13.2.1 on ARM. I could not reproduce on macOS Big Sur 11.7.6 on x86 though.

I pushed a commit to find the Brew LZMA and update `CFLAGS` accordingly: https://github.com/pwalch/lonesnake/commit/d53f31bdbdd2121842d33557bafc2495812ce4fa. Would you mind letting me know if it fixes it for you?

**Comment by sansuiso** [comment]
Thank you very much! I have tested locally with python 3.10/3.11 and I was able to `import lzma` with both without any problem. The fix works for me too 👍

**Comment by pwalch** [comment]
Thanks for the confirmation! I merged the fix and released the new version 0.20.0 :balloon: You can install it with `brew install lonesnake`.
